### ISSUE TITLE
Autogenerated count query fails for native query

### ISSUE THREAD

**Issue by domgom** [spring-projects/spring-data-jpa#2802]
```java
public interface WalletTransactionRepository extends CrudRepository<WalletTransaction, Long> {
 @Query(
            value = "SELECT * FROM wallet_transactions wt"
                    + " WHERE 1=1"
                    + " AND (:walletId = 0 OR wt.wallet_id = :walletId)"
                    + " AND (coalesce(:transactionTypes) IS NULL OR wt.transaction_type IN (:transactionTypes))"
                    + " AND (:createdAfter IS NULL OR wt.created > :createdAfter)",
            nativeQuery = true
)
    Page<WalletTransaction> findByWalletIdAndTransactionTypeAndCreatedAfter(@Param("walletId") long walletId,
                                                                            @Param("transactionTypes") Set<String> transactionTypes, @Param("createdAfter") ZonedDateTime createdAfter,
Pageable pageable);
}
```
This query fails but works fine changing the return type to List<WalletTransaction>.

The problem is that the count query is using the table name as a field, with the executed sql being something like this:
```
SELECT count(wt) FROM wallet_transactions wt
WHERE 1=1
AND (1 = 0 OR wt.wallet_id = 1)
AND (coalesce('DEPOSIT','WITHDRAWAL')
 IS NULL OR wt.transaction_type IN ('DEPOSIT','WITHDRAWAL'))
AND (NULL IS NULL OR wt.created > NULL);
```
If I add a countQuery attribute using `SELECT count(*)` instead of the autogenerated `SELECT count(wt)` it works again.

```
Spring boot version 3.0.0
Spring data version 3.0.0
Java 19
org.hibernate.dialect.MySQL8Dialect
docker mysql:8.0
```

**Comment by mp911de** [comment]
That should be fixed with #2773. Can you test against Spring Data JPA 3.0.2 snapshots to verify the fix is working for you?

**Comment by domgom** [comment]
All good.
I can see the in the logs the correctly generated sql for the count:
`2023-02-17 10:29:31.520 DEBUG [main] org.hibernate.SQL - select count(1) FROM wallet_transactions wt` [...]

cheers 🍾
